src/com/google/javascript/jscomp/parsing/parser/Scanner.java

```diff
@@ -901,8 +901,65 @@ private static boolean isIdentifierStart(char ch) {
         | (ch >= 0x03B1 & ch <= 0x03C9); // Greek lowercase letters
   }
 
+  // Check if char is Unicode Category "Combining spacing mark (Mc)"
```

Comment:
Not compliant (see #3647), per spec `UnicodeCombiningMark` also includes "Non-spacing mark (Mn)" category, but this addresses #3639 and a decent scope of similar possibilities with as minimal change as possible.